pkg/standalone-utils/test/LidoRelayer.test.ts

```diff
@@ -1,462 +1,950 @@
 
 describe('LidoRelayer', function () {
-  let WETH: Token, wstETH: Token;
+  let stETH: Token, wstETH: Token;
   let basePoolId: string;
-  let basePoolTokens: TokenList;
-  let sender: SignerWithAddress, admin: SignerWithAddress;
+  let tokens: TokenList;
+  let sender: SignerWithAddress, recipient: SignerWithAddress, admin: SignerWithAddress;
   let vault: Vault, basePool: StablePool;
-  let relayer: Contract;
+  let relayer: Contract, relayerLibrary: Contract;
 
   before('setup signer', async () => {
-    [, admin, sender] = await ethers.getSigners();
+    [, admin, sender, recipient] = await ethers.getSigners();
   });
 
   sharedBeforeEach('deploy Vault', async () => {
+    const [deployer] = await ethers.getSigners();
     vault = await Vault.create({ admin });
 
-    const wethContract = await deployedAt('TestWETH', await vault.instance.WETH());
-    WETH = new Token('WETH', 'WETH', 18, wethContract);
+    const stETHContract = await deploy('MockStETH', { args: [deployer.address, 'stETH', 'stETH', 18] });
+    stETH = new Token('stETH', 'stETH', 18, stETHContract);
 
-    const wstETHContract = await deploy('MockWstETH', { args: [WETH.address] });
+    const wstETHContract = await deploy('MockWstETH', { args: [stETH.address] });
     wstETH = new Token('wstETH', 'wstETH', 18, wstETHContract);
-
-    relayer = await deploy('LidoRelayer', { args: [vault.address, wstETH.address] });
   });
 
   sharedBeforeEach('deploy pool', async () => {
-    basePoolTokens = new TokenList([WETH, wstETH]).sort();
+    tokens = new TokenList([stETH, wstETH]).sort();
 
-    basePool = await StablePool.create({ tokens: basePoolTokens, vault });
+    basePool = await StablePool.create({ tokens, vault });
     basePoolId = basePool.poolId;
 
     // Seed liquidity in pool
-    await WETH.mint(admin, fp(200));
-    await WETH.approve(vault.address, MAX_UINT256, { from: admin });
+    await stETH.mint(admin, fp(200));
+    await stETH.approve(vault.address, MAX_UINT256, { from: admin });
 
-    await WETH.mint(admin, fp(150));
-    await WETH.approve(wstETH.address, fp(150), { from: admin });
+    await stETH.mint(admin, fp(150));
+    await stETH.approve(wstETH.address, fp(150), { from: admin });
     await wstETH.instance.connect(admin).wrap(fp(150));
     await wstETH.approve(vault.address, MAX_UINT256, { from: admin });
 
     await basePool.init({ initialBalances: fp(100), from: admin });
   });
 
   sharedBeforeEach('mint tokens to sender', async () => {
-    await WETH.mint(sender, fp(100));
-    await WETH.approve(vault.address, fp(100), { from: sender });
+    await stETH.mint(sender, fp(100));
+    await stETH.approve(vault.address, fp(100), { from: sender });
 
-    await WETH.mint(sender, fp(2500));
-    await WETH.approve(wstETH.address, fp(150), { from: sender });
+    await stETH.mint(sender, fp(2500));
+    await stETH.approve(wstETH.address, fp(150), { from: sender });
     await wstETH.instance.connect(sender).wrap(fp(150));
```

Comment:
Let's make this pool have a third token: it is very weird that we are doing 1) wrap steth into wsteth 2) swap wsteth for steth, making the tests harder to read. With a third token we'll always be able to reference either wsteth or steth, plus the neutral third one when required.

Reply:
Ah, good catch. I just did a find-and-replace when giving a proper stETH implementation and didn't think about how we were using WETH in the pool as well so it would make the pool weird.